radixdlt/src/main/java/com/radixdlt/middleware2/network/SimpleEventCoordinatorNetwork.java

```diff
@@ -106,13 +124,43 @@ public void sendVote(Vote vote, ECPublicKey leader) {
 		}
 	}
 
-	private void send(Message message, ECPublicKey recipient) {
+	@Override
+	public Single<Vertex> getVertex(Hash vertexId, ECPublicKey node) {
+		if (this.selfPublicKey.equals(node)) {
+			throw new IllegalStateException("Should never need to retrieve a vertex from self.");
+		}
+
+		return Single.create(emitter -> {
+			final Optional<Peer> peer = this.addressBook.peer(node.euid());
+			if (!peer.isPresent()) {
+				emitter.onError(new RuntimeException(String.format("Peer with pubkey %s not present", node)));
```

Comment:
Should have a more specific exception here, rather than the general RuntimeException.

IllegalStateException would be fine.